tests/ga/test_extension.py

```diff
@@ -2218,38 +2218,50 @@ def test_ext_handler_with_cgroup_enabled(self, *args):
         exthandlers_handler.run()
         self._assert_no_handler_status(protocol.report_vm_status)
 
+    @skip_if_predicate_true(is_trusty_in_travis, "Does not run on Trusty in Travis as CPU cgroup is not mounted")
+    @patch('azurelinuxagent.common.event.EventLogger.add_metric')
     @patch('azurelinuxagent.common.event.EventLogger.add_event')
     @attr('requires_sudo')
-    def test_ext_handler_and_monitor_handler_with_cgroup_enabled(self, patch_add_event, *args):
+    def test_ext_handler_and_monitor_handler_with_cgroup_enabled(self, patch_add_event, patch_add_metric, *args):
         self.assertTrue(i_am_root(), "Test does not run when non-root")
 
-        test_data = WireProtocolData(DATA_FILE)
-        exthandlers_handler, monitor_handler, protocol = self._create_mock(test_data, *args)
+        # This test has some timing issues when systemd is managing cgroups, so we force the file system API
```

Comment:
I added this comment/code to 1 single test as a temporary workaround, but now the same hack is being propagated to other tests... can we handle this better?

Reply:
I'll sync on this offline. Thanks.

Reply:
Added a suggestion above